### DISCUSSION TITLE
Is there a way to generate deterministic class names?

### DISCUSSION THREAD

**Discussion by kevinxh** [chakra-ui/chakra-ui#3653] [Q&A]
Hi Chakra Community!

I'm wondering if there is a way to generate deterministic class name instead of having random class names like:
<img width="266" alt="Screen Shot 2021-03-23 at 2 27 07 PM" src="https://user-images.githubusercontent.com/10948652/112222142-8fb97600-8be5-11eb-80b7-d1a11b23e406.png">

Asking because it would be nice to use the class name for testing purposes.

Thank you,

Kevin

**Comment by chasinhues** [comment]
Install this package and configure jest to use the snapshot serializer

https://github.com/emotion-js/emotion/tree/master/packages/jest#snapshot-serializer

I haven't done this before but apparently you can also replace the classnames as desired: https://github.com/emotion-js/emotion/tree/master/packages/jest#classnamereplacer

**Reply by KatFishSnake** [reply]
@chasinhues am i missing something?, its not for jest specifically but for cypress and debugging
Does chakra supports something like that?

**Reply by bharat-dussa** [reply]
if you are testing, i would recommend to use data-testid,and id for the same, it would be better by this you do not need to be worry about classnames

**Comment by naquarius** [comment]
Running into the similar issue. every time there is a small change it's always the classname that gets updated, would be awesome if we can actually use deterministic class names.

**Reply by roitalpaz** [reply]
any luck with this?


Found this: 
https://emotion.sh/docs/@emotion/babel-plugin#autolabel
but cant seem to make it work

**Reply by afraser** [reply]
Same here for `autolabel`. The issue is likely that the styles are not being created by directly using `css()` or `styled()` https://emotion.sh/docs/@emotion/babel-plugin#labelformat
